### PR TITLE
Add release staging builds to the pipeline

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

File: tools/pkgrepo.py [hunk 1]
```diff
@@ -821,20 +826,529 @@ def staging(ctx: Context, repo_path: pathlib.Path, rc_build: bool = False):
...
+    if rc_build:
+        bucket_folder = "salt_rc/py3"
+    else:
+        bucket_folder = "salt/py3"
+
+    files_to_copy: list[str]
+    files_to_delete: list[str] = []
+    files_to_duplicate: list[tuple[str, str]] = []
+
+    ctx.info("Grabing remote file listing of files to copy...")
```
Comment (twangboy) [inline tools/pkgrepo.py:913]: grabbing

File: tools/pkgrepo.py [hunk 2]
```diff
@@ -821,20 +826,529 @@ def staging(ctx: Context, repo_path: pathlib.Path, rc_build: bool = False):
...
+    files_in_backup: dict[str, datetime] = {}
+    files_to_backup: list[tuple[str, datetime]] = []
+
+    ctx.info("Grabing remote listing of files in backup ...")
+    for entry in _get_repo_detailed_file_list(
+        bucket_name=tools.utils.BACKUP_BUCKET_NAME,
+    ):
+        files_in_backup[entry["Key"]] = entry["LastModified"]
+
+    ctx.info("Grabing remote listing of files to backup ...")
```
Comment (twangboy) [inline tools/pkgrepo.py:843]: grabbing

File: tools/pkgrepo.py [hunk 3]
```diff
@@ -821,20 +826,529 @@ def staging(ctx: Context, repo_path: pathlib.Path, rc_build: bool = False):
...
 
+@repo.command(name="backup-previous-releases")
+def backup_previous_releases(ctx: Context):
+    """
+    Backup previous releases.
+    """
+    files_in_backup: dict[str, datetime] = {}
+    files_to_backup: list[tuple[str, datetime]] = []
+
+    ctx.info("Grabing remote listing of files in backup ...")
```
Comment (twangboy) [inline tools/pkgrepo.py:837]: grabbing

File: tools/release.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    help="Release Related Commands",
+    description=__doc__,
+)
+
+
+@release.command(
+    name="upload-artifacts",
+    arguments={
+        "salt_version": {
+            "help": ("The salt version to release. "),
```
Comment (twangboy) [inline tools/release.py:41]: Do we need the space after the period? `release. "),`
And, why is this one wrapped in parenthesis when the others aren't?

**Review by twangboy — CHANGES_REQUESTED** [review]

File: tools/pkgrepo.py (hunk 1 above)
Comment (s0undt3ch) [reply tools/pkgrepo.py]: Fixed

File: tools/pkgrepo.py (hunk 3 above)
Comment (s0undt3ch) [reply tools/pkgrepo.py]: Fixed

File: tools/release.py (hunk 4 above)
Comment (s0undt3ch) [reply tools/release.py]: Fixed. Thanks!

File: tools/pkgrepo.py (hunk 2 above)
Comment (s0undt3ch) [reply tools/pkgrepo.py]: Fixed

**Review by whytewolf — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by dmurphy18 — DISMISSED** [review]
Left comments about spaces in names, but got over my old-C style discomfort about it, so disregard if feel so, but didn't go back and delete since so many files. What was that about smaller PRs :laughing: 

The other questions are more to do with architecture support, the arm64 for Apple, understand if not there because not provided to test on yet

**Review by dwoz — DISMISSED** [review]

**Review by cmcmarrow — DISMISSED** [review]

**Pull request merged**